File: mythicdrops/src/test/java/com/tealcube/minecraft/bukkit/mythicdrops/templating/RandSignTemplateTest.java
package com.tealcube.minecraft.bukkit.mythicdrops.templating;

import org.junit.Assert;
import org.junit.Test;

public class RandSignTemplateTest {

    @Test
    public void doesApplyReturnAboutHalfPlus() {
        RandSignTemplate randSignTemplate = new RandSignTemplate();
        String templateToTest = "%randsign%";

        int amount = 0;
        for (int i = 0; i < 100; i++) {
            String val = randSignTemplate.apply(templateToTest);
            amount += (val.equalsIgnoreCase("+") ? 1 : 0);
        }

        Assert.assertTrue(amount > 35);
    }

}

File: mythicdrops/src/test/kotlin/com/tealcube/minecraft/bukkit/mythicdrops/templating/RandSignTemplateTest.kt
/*
 * This file is part of MythicDrops, licensed under the MIT License.
 *
 * Copyright (C) 2019 Richard Harrah
 *
 * Permission is hereby granted, free of charge,
 * to any person obtaining a copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
 * and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
 * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
 * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
package com.tealcube.minecraft.bukkit.mythicdrops.templating

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class RandSignTemplateTest {
    @Test
    fun `does invoke return appropriate amounts of "+" and "-"`() {
        var returnedMinuses = 0
        var returnedPluses = 0
        repeat(100) {
            if (RandSignTemplate.invoke("") == "+") {
                returnedPluses++
            } else {
                returnedMinuses++
            }
        }
        assertThat(returnedMinuses).isGreaterThan(35)
        assertThat(returnedPluses).isGreaterThan(35)
    }
}
